keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/admin/CtUserResource.java

```diff
@@ -0,0 +1,195 @@
+     * be no link back to click after actions have completed.  Redirect uri must be a valid uri for the
+     * particular clientId.
+     *
+     * @param redirectUri    Redirect uri
+     * @param clientId       Client id
+     * @param lifespan       Number of seconds after which the generated token expires
+     * @param themeRealmName Name of the realm used for theming
+     * @param actions        required actions the user needs to complete
+     * @return
+     */
+    @Path("execute-actions-email")
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response executeActionsEmail(@QueryParam(OIDCLoginProtocol.REDIRECT_URI_PARAM) String redirectUri,
+                                        @QueryParam(OIDCLoginProtocol.CLIENT_ID_PARAM) String clientId,
+                                        @QueryParam("lifespan") Integer lifespan,
+                                        @QueryParam("themeRealm") String themeRealmName,
+                                        List<String> actions) {
+        auth.users().requireManage(user);
+
+        if (user.getEmail() == null) {
+            return ErrorResponse.error("User email missing", Response.Status.BAD_REQUEST);
+        }
+
+        if (!user.isEnabled()) {
+            throw new WebApplicationException(
+                    ErrorResponse.error("User is disabled", Response.Status.BAD_REQUEST));
+        }
+
+        if (redirectUri != null && clientId == null) {
+            throw new WebApplicationException(
+                    ErrorResponse.error("Client id missing", Response.Status.BAD_REQUEST));
+        }
+
+        if (clientId == null) {
+            clientId = Constants.ACCOUNT_MANAGEMENT_CLIENT_ID;
+        }
+
+        ClientModel client = realm.getClientByClientId(clientId);
+        if (client == null) {
+            logger.debugf("Client %s doesn't exist", clientId);
+            throw new WebApplicationException(
+                    ErrorResponse.error("Client doesn't exist", Response.Status.BAD_REQUEST));
+        }
+        if (!client.isEnabled()) {
+            logger.debugf("Client %s is not enabled", clientId);
+            throw new WebApplicationException(
+                    ErrorResponse.error("Client is not enabled", Response.Status.BAD_REQUEST));
+        }
+
+        String redirect;
+        if (redirectUri != null) {
+            redirect = RedirectUtils.verifyRedirectUri(session, redirectUri, client);
+            if (redirect == null) {
+                throw new WebApplicationException(
+                        ErrorResponse.error("Invalid redirect uri.", Response.Status.BAD_REQUEST));
+            }
+        }
+
+        if (!StringUtils.isBlank(themeRealmName)) {
+            RealmManager realmManager = new RealmManager(session);
+            RealmModel themeRealm = realmManager.getRealmByName(themeRealmName);
+            if (themeRealm == null) {
+                return ErrorResponse.error("Invalid realm name", Response.Status.NOT_FOUND);
```

Comment:
For consistency with the others errors (i.e. client does not exists), maybe we could use BAD_REQUEST